src/lib.js

```diff
@@ -252,6 +256,15 @@
 
     info(options,
       `Sending message to target window (event: ${name}) -->`, parsedData);
+
+    if (options.queue) {
+      info(options, 'Queueing message in case target window is not ready');
+      queueHandler = on('target:loaded', () => {
+        queueHandler && queueHandler.off();
+        target.postMessage(parsedData, origin);
```

Comment:
That might be a false positive because the default origin is `*`. Will do additional tests on different origins to make sure this is not a security hole.